### ISSUE TITLE
Create tables for categories, sites

### ISSUE THREAD

**Issue by k88hudson** [oyiptong/splice#222]
`categories`

Column    | Type         | Modifiers
------------- | ------------- | -------------
id              | integer     | not null
name        | character varying(255) | not null
archived   | boolean | not null default false
created_at | timestamp | not null now()

`sites`

Column    | Type         | Modifiers
------------- | ------------- | -------------
id              | integer     | not null
url        | character varying(255) | not null
archived   | boolean | default false
created_at | timestamp | now()

`sites_categories`

Column    | Type         | Modifiers
------------- | ------------- | -------------
site_id | integer | not null
category_id | integer | not null

**Comment by k88hudson** [comment]
@mzhilyaev @tspurway  could you take a look at this and see if it makes sense?

**Comment by oyiptong** [comment]
Perhaps we could add a description column so that categories could provide more context to an end-user.

**Comment by ncloudioj** [comment]
Good catch! @oyiptong. In addition to `explanation`, we also need a field `adgroup_name`. Firefox needs this to fill up the explanation template. For example, "A {adgroup_name} site suggestion". Note that it's different from the `name` field. e.g. `name`: "Sports_General", `adgroup_name`: "sports".

**Comment by mzhilyaev** [comment]
few questions:

- Is the site url == site domain?
- Is there going to be a single site table, accessed by bucketer population scripts, or do you see the bucketer having a different table where it keeps domains, ranks, titles, etc... ?

**Comment by k88hudson** [comment]
@oyiptong @mzhilyaev good suggestions!

@mzhilyaev:
* yes, site url is domain (TLD +1)
* buckets sometimes contain domains/sites that do *not* exist in our data sources; also, if sites get removed in the source data, we don't want to automatically edit our categories, so I think it does need to be separate. The source data should be read-only, since we're replacing it/updating it periodically
